backends/edk2-compat/edk2-svc-generate.c

```diff
@@ -293,20 +296,10 @@ static int parse_opt(int key, char *arg, struct argp_state *state)
 				rc = ALLOC_FAIL;
 				break;
 			}
-			// make sure -t flag has two strings <date> <time>
-			if (state->next >= state->argc || state->argv[state->next][0] == '-') {
-				prlog(PR_ERR, "ERROR: Incorrect flag '-t', see usage...\n");
-				argp_usage(state);
-				rc = ARG_PARSE_FAIL;
-				break;
-			}
-			// make sure timestamp is correct format
-			if (sscanf(arg,"%hd-%hhd-%hhd", &args->time->year, &args->time->month, &args->time->day) != 3
-				|| sscanf(state->argv[state->next], "%hhd:%hhd:%hhd", &args->time->hour, &args->time->minute, &args->time->second) != 3) {
-				prlog(PR_ERR, "ERROR: Could not parse given timestamp, make sure it is in format 'y-m-d h:m:s'\n");
-				rc = ARG_PARSE_FAIL;
-			}
-			state->next++;
+            if (parseCustomTimestamp(args->time, arg)) {
+                prlog(PR_ERR, "ERROR: Could not parse given timestamp %s, make sure it is in format <YYYY-MM-DDThh:mm:ss\n", arg);
```

Comment:
Not sure what the `<` in the message is supposed to indicate.
I see the help string around a lot, might be worth factoring that out to a macro in a header somewhere so if we have to change the timestamp (which hopefully we don't...), we don't have to change this everywhere.